macrosynergy/download/dataquery.py

```diff
@@ -33,212 +41,234 @@
+                requests.exceptions.InvalidURL,
+                requests.exceptions.InvalidSchema,
+                requests.exceptions.ChunkedEncodingError,
+            ]
+
+            if any([isinstance(exc, e) for e in known_exceptions]):
+                logger.warning(error_statement)
+                retry_count += 1
+                time.sleep(API_DELAY_PARAM)
+            else:
+                raise exc
+
+    raise DownloadError(
+        f"Request to {log_url} failed with status code {response.status_code}. "
+        "No longer retrying."
+    )
 
 
 class OAuth(object):
-    """Accessing DataQuery via OAuth.
+    """
+    Class for handling OAuth authentication for the DataQuery API.
 
-    :param <str> client_id: string with client id, username.
-    :param <str> client_secret: string with client secret, password.
-    :param <str> url:
-    :param <str> token_url:
-    :param <str> dq_resource_id:
+    :param <str> client_id: client ID for the OAuth application.
+    :param <str> client_secret: client secret for the OAuth application.
+    :param <dict> proxy: proxy to use for requests. Defaults to None.
+    :param <str> base_url: base URL for OAuth access.
+    :param <str> token_url: URL for getting OAuth tokens.
+    :param <str> dq_resource_id: resource ID for the JPMaQS Application.
+
+    :return <OAuth>: OAuth object.
+
+    :raises <ValueError>: if client_id or client_secret are not strings.
+    :raises <Exception>: other exceptions may be raised by underlying functions.
     """
 
     def __init__(
         self,
         client_id: str,
         client_secret: str,
-        url: str = OAUTH_BASE_URL,
+        proxy: Optional[dict] = None,
+        base_url: str = OAUTH_BASE_URL,
         token_url: str = OAUTH_TOKEN_URL,
         dq_resource_id: str = OAUTH_DQ_RESOURCE_ID,
-        proxy: Optional[dict] = None,
     ):
+        try:
+            id_error = f"client_id argument must be a <str> and not {type(client_id)}."
+            assert isinstance(client_id, str), id_error
+            secret_error = (
+                f"client_secret must be a <str> and not {type(client_secret)}."
+            )
+            assert isinstance(client_secret, str), secret_error
+            proxy_error = f"proxy must be a <dict> and not {type(proxy)}."
+            assert isinstance(proxy, dict) or proxy is None, proxy_error
+            url_error = f"base_url must be a <str> and not {type(base_url)}."
+            assert isinstance(base_url, str), url_error
+            token_url_error = f"token_url must be a <str> and not {type(token_url)}."
+            assert isinstance(token_url, str), token_url_error
+            dq_resource_id_error = (
+                f"dq_resource_id must be a <str> and not {type(dq_resource_id)}."
+            )
+            assert isinstance(dq_resource_id, str), dq_resource_id_error
 
-        self.base_url: str = url
+        except AssertionError as exc:
+            raise ValueError(exc)
```

Comment:
Similar to other comments, this should either be an assert error or a type error, and not be inside a try-catch block.